### PR TITLE
CONN-1759 ACP/IACP AuthzDecisionStatement

### PR DESCRIPTION
**Pull request by trantang**
One of the ACP or IACP is required in the AuthzDecisionStatement; if none exist AuthzDecisionStatement is invalid and should not be created.

### CODE REVIEW

**Review by ptambellini01 — CHANGES_REQUESTED** [review]
In the HOKSamlAssertionBuilder, there are only one condition being tested, whether or not there are two conditions ACP/IACP or not. Shouldn't it at least have one other test. Whether it fails to write if there aren't at least one condition.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilder.java [hunk 1]
```diff
@@ -496,21 +500,21 @@ private static DateTime setEndValidTime(DateTime endValidTimeArg, DateTime issue
...
+        final List evidenceInstanceAccessConsentValues = checkHcidPrefixInList(properties.getEvidenceInstantAccessConsent());
 
         return createEvidenceStatements(accessConstentValues, evidenceInstanceAccessConsentValues);
     }
 
     public List<AttributeStatement> createEvidenceStatements(final List accessConstentValues,
         final List evidenceInstanceAccessConsentValues) {
         List<AttributeStatement> statements;
-        if (accessConstentValues == null) {
+        if (CollectionUtils.isEmpty(accessConstentValues)) {
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilder.java]: These 2 if conditions and log statements are unnecessary. We can rew3ord and write these log statement in OpenSaml2Builder.createEvidenceStatements method.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/OpenSAML2ComponentBuilder.java [hunk 2]
```diff
@@ -518,10 +518,10 @@ public Evidence createEvidence(final List<Assertion> assertions) {
         final List evidenceInstanceAccessConsentValues, final String namespace) {
         List<AttributeStatement> statements = new ArrayList<>();
         final List<Attribute> attributes = new ArrayList<>();
-        if (accessConstentValues != null) {
+        if (CollectionUtils.isNotEmpty(accessConstentValues)) {
             attributes.add(createAttribute(null, NhincConstants.ACCESS_CONSENT_ATTR, namespace, accessConstentValues));
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/OpenSAML2ComponentBuilder.java]: ACCESS_CONSENT_ATTR and INST_ACCESS_CONSENT_ATTR are defined in both SamlConstants and NhincConstants. Remove from NhincConstants.

**Review by TabassumJafri — CHANGES_REQUESTED** [review]
Please see my inline comments

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilder.java [hunk 3]
```diff
@@ -757,4 +753,28 @@ public static String checkPrefixBeforeAppend(final String checkValue, final Stri
             return checkValue;
         }
     }
+
+    public static List<Object> checkPrefixInList(final List<Object> valueList, final String checkPrefix){
```
Comment (jsmith2) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilder.java]: These two methods can be combined.   No need to create a whole new method just to pass in a constant.

**Review by ptambellini01 — APPROVED** [review]

**Review by mhpnguyen — APPROVED** [review]

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilder.java [hunk 4]
```diff
@@ -757,4 +753,24 @@ public static String checkPrefixBeforeAppend(final String checkValue, final Stri
...
+        return tempList;
+    }
+
+    protected boolean isAcpOrIacpExists(final CallbackProperties properties) {
+        final List accessConstentValues = properties.getEvidenceAccessConstent();
+        final List evidenceInstanceAccessConsentValues = properties.getEvidenceInstantAccessConsent();
+
+        //return false if ACP/IACP does not exist
+        //only created the AuthzDecisionStatement if true
+        if(CollectionUtils.isEmpty(accessConstentValues) && CollectionUtils.isEmpty(evidenceInstanceAccessConsentValues)){
```
Comment (jsmith2) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilder.java]: This should have popped up on sonarqube.  Can be simplified to:

return !Collections.IsEmpty(accessConsentValues) && !CollectionUtils.isEmpty(evidenceInstanceAccessConsentValues);

No need for if statement.

**Review by jsmith2 — APPROVED** [review]

**Pull request merged**